docs/src/pages/customization/default-theme/default-theme.md

```diff
@@ -24,3 +24,5 @@ To remove the need for injecting a theme in the React's context **systematically
 - `@material-ui/core/styles/makeStyles` wraps `@material-ui/styles/makeStyles`.
 - `@material-ui/core/styles/withStyles` wraps `@material-ui/styles/withStyles`.
 - `@material-ui/core/styles/styled` wraps `@material-ui/styles/styled`.
+
+For practical usage (such as styling a Material-UI component instance), please import the style modules (`makeStyles`, `withStyles` and `styled`) from `@material-ui/core/styles`.
```

Comment:
I would hesitate to use the word practical here. I'm cc'ing @oliviertassinari for visibility :)

Reply:
@joshwooding that sounds pretty good. ill wait to get a bit more feedback before posting a new change.

Reply:
I wouldn't consider this practical either. If you create a custom theme then you don't want to import from `core/styles` because that will always bundle default styles. This is only practical for people that can't test their components in their app context.

Reply:
That makes sense, I wonder if that'll confuse some people. @bigtone1284 Ignore what I said :P 

What do you think @oliviertassinari

Reply:
@eps1lon point is valid. I'm probably too biased on this matter to have an objective point of view. I believe the original issue is about minimizing people confused about the two modules. I think that this change is off the road, it would have a minimal impact. I believe people don't find this section in the first place.

Reply:
@oliviertassinari Are you suggesting then that this section being linked to from somewhere in the docs that people commonly find themselves in that is related to using one of the two modules?

Reply:
I believe that people don't find this information in the first place. It's a difficulte problem. I don't have any good answer.

@eps1lon, you had some idea on how to improve the situation. Maybe you would want to lead this change?

Reply:
From what I can tell the recommendation is as simple as "custom theme => mui/styles, default theme => mui/core/styles". 

Though you have to be aware that switching from default theme to custom theme will require changing your test setup.

I think the underlying issue is that people just insist on shallow rendering or traversing the react tree for assertions. We can't improve the situation for those people without severly limiting our ability to ship features.

IMO the mui/styles should be hidden deep down for people who want next level customization and know that bundling the default theme is bad for perf. It isn't something beginners need to worry about.

Reply:
@eps1lon Wow great input on this. So in terms of actionable ways to improve this....

1) I can change the existing default-theme.md docs to say something like:

**If you are using the default theme, import the style modules (`makeStyles`, `withStyles` and `styled`) from `@material-ui/core/styles`, as the default-theme is bundled with these methods (we recommend this or most beginners using mui). If you are using a custom theme, import the style modules from `@material-ui/styles`, but be advised that doing so requires changing your testing setup.**

As a follow up, are their docs on changing the testing setup for this case?

Reply:
There aren’t any docs as far as I know but it doesn’t really need you to change your test setup too much I believe you just need to test them under a theme provider. You could probably just add it under the above paragraph.

Reply:
It doesn't seem this content belongs in this section to begin with.  Some alternatives:

Styles section?
FAQ?
Testing guide?

As to the content – we document @material-ui/styles as the styling solution of choice, so:
- When would a developer want to use @material-ui/core/styles instead?
- What are the implications of doing so?

Reply:
> "custom theme => mui/styles, default theme => mui/core/styles".

@eps1lon What do you think of: using material-ui/core => mui/core/styles, not using core mui/styles?
The default style is already bundled by the core components.

Reply:
> When would a developer want to use @material-ui/core/styles instead?

This was the only way in v3. It was kept to minimize the number of BCs and more importantly, to not require the presence of a ThemeProvider at the top of the tree. It optimizes for the onboarding.

Reply:
> This was the only way in v3. It was kept to minimize the number of BCs

Which is fine in of itself - if a dev is using it since v3, they can keep using it; but (trying to) document it as an alternative to the standalone styles package is the cause of confusion.

> more importantly, to not require the presence of a ThemeProvider at the top of the tree

...which is an important internal implementation detail, but it doesn't answer the question "When would a developer want to use @material-ui/core/styles instead?".

Reply:
> "When would a developer want to use @material-ui/core/styles instead?"

When the developer doesn't want to have to worry about having a theme provider as a parent. It breaks without the theme provider when using mui/styles directly.

Reply:
> It breaks without the theme provider when using mui/styles directly

What breaks? We demonstrate it here: https://material-ui.com/styles/basics/

Reply:
If we are suggesting `@material-ui/core` users don’t use `@material-ui/styles` we need to differentiate the documentation more. `Styles` is too generic in the current case. Maybe we should split the drawer up with Dividers with sub headers?

Reply:
> What breaks?

@mbrookes When working on styles v3-alpha we had a vague of issues related to `theme.whatever()` accesses breaking. People copy and past the demos, demos that assume a rich theme structure is present. It's not the case when importing from mui/styles they might be nothing (it will warn you)

Reply:
@oliviertassinari Got it, thanks.

So we either need to remove theme from the demos, or as @joshwooding do something with the `/styles` documentation. I would favour moving it to the package directory. We could still cross-reference it in the Material-UI docs, it just wouldn't form part of them.

Reply:
@mbrookes @oliviertassinari so it feels like maybe removing theme from demos makes sense as lots of users are just dropping into the demo docs looking for an example they can use almost out of the box? Is that what I am hearing here?

Reply:
@oliviertassinari What are your thoughts on moving the `/styles` documentation?

Reply:
In the long term, I think that we should bury `/styles` down. As of today, I believe the last majority of our users have to face it, at least once when using the library: when they copy & paste the demos. For this reason, I think that we should keep it up. I would focus on better explaining the diff. Maybe with a FAQ question? It's not that incredibly important. People wonder why? People see the theme is missing, have a warning, add the provider.